### ISSUE TITLE
Dev Mode Continuous Testing: Arc fails in Maven Multi-Module project with "TestHelper" module.

### ISSUE THREAD

**Issue by serahl** [quarkusio/quarkus#23311]
### Describe the bug

We use Quarkus in a multi-module Maven project setup.
When we introduced a module thats only used in the test Scope (as a Util/Helper module), we noticed that our Continuous Testing broke.
I simplified the setup in an example-project to recreate the error (see attached .zip or [Repo](https://github.com/serahl/quarkus-cont-testing-arc-error-example))

We have the following modules:

- dbservice
- testhelper
  - depends on dbservice, implements a decorator for a service interface from dbservice
- restservice
  - depends on dbservice
  - depends on testhelper only for the test scope.

When running `mvn clean test`, everything works fine the tests are green.
However, when starting the restservice module in dev mode via `mvn clean package -am -pl restservice -P devmode-restservice` or the standalone quarkus:test mode via `mvn clean package -am -pl restservice -P testmode-restservice` and enabling continuous testing, we run into an error.
 

### Expected behavior

The @QuarkusTest testcases should be run successfully.

### Actual behavior

Quarkus Arc throws an exception. It looks like the ClassLoader of the Continuous Testing does not pick up the dependencies of the TestHelper module correctly.

2022-01-31 15:21:19,440 ERROR [io.qua.test] (Test runner thread) Test GreetingResourceTest#testHelloEndpoint() failed
: java.lang.RuntimeException: java.lang.ExceptionInInitializerError
        at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:587)
        at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:660)
        at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
        ...
        at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:60)
        at io.quarkus.deployment.dev.testing.JunitTestRunner$3.run(JunitTestRunner.java:228)
        at io.quarkus.deployment.dev.testing.ModuleTestRunner$2.run(ModuleTestRunner.java:90)
        at io.quarkus.deployment.dev.testing.TestSupport.runInternal(TestSupport.java:387)
        at io.quarkus.deployment.dev.testing.TestSupport$2.run(TestSupport.java:296)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:467)
        at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:221)
        at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:245)
        at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:564)
        at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:602)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$10(ClassBasedTestDescriptor.java:381)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAlcks(ClassBasedTestDescriptor.java:381)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:205)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:80)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
        ... 34 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        ... 46 more
Caused by: java.lang.RuntimeException: Failed to initialize Arc
        at io.quarkus.arc.Arc.initialize(Arc.java:26)
        at io.quarkus.arc.runtime.ArcRecorder.getContainer(ArcRecorder.java:40)
        at io.quarkus.deployment.steps.ArcProcessor$generateResources1025303321.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.ArcProcessor$generateResources1025303321.deploy(Unknown Source)
        ... 47 more
Caused by: java.lang.NoClassDefFoundError: de/serahl/quarkustest/dbservice/DbService
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:445)
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:405)
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:455)
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:405)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:467)
        at de.serahl.quarkustest.testhelper.DbServiceDecorator_Bean.<init>(Unknown Source)
        at io.quarkus.arc.setup.Default_ComponentsProvider.addBeans1(Unknown Source)
        at io.quarkus.arc.setup.Default_ComponentsProvider.getComponents(Unknown Source)
        at io.quarkus.arc.impl.ArcContainerImpl.<init>(ArcContainerImpl.java:118)
        at io.quarkus.arc.Arc.initialize(Arc.java:20)
        ... 50 more
Caused by: java.lang.ClassNotFoundException: de.serahl.quarkustest.dbservice.DbService
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:429)
        at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:405)
        ... 63 more

### How to Reproduce?

Seee linked repository or attached file.
[quarkus-cont-testing-arc-error-example.zip](https://github.com/quarkusio/quarkus/files/7971479/quarkus-cont-testing-arc-error-example.zip)

### Output of `java -version`

openjdk version "17.0.1" 2021-10-19 LTS OpenJDK Runtime Environment Microsoft-28056 (build 17.0.1+12-LTS) OpenJDK 64-Bit Server VM Microsoft-28056 (build 17.0.1+12-LTS, mixed mode, sharing)

### Quarkus version or git rev

2.6.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.3 (ff8e977a158738155dc465c6a97ffaf31982d739)

**Comment by famod** [comment]
Thanks for reporting. This is fixed in 2.7.0.Final, most likely via some extensive change by @aloubyansky.

Full platform release will be on February 2nd. If you don't want to wait for that you can use
```xml
<quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
```
instead of
```
<quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
```
PS: I noticed that you are using `jandex-maven-plugin`. Make sure to align it with what's in the BOM: https://github.com/quarkusio/quarkus/blob/main/bom/application/pom.xml#L21

**Comment by serahl** [comment]
Looking forward to it!
